spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAlterSortOrder.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.extensions;
+
+import java.util.Map;
+import org.apache.iceberg.Table;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestAlterSortOrder extends SparkExtensionsTestBase {
```

Comment:
I don't think this is needed. We should avoid adding tests are are specific to Spark, but only test changes to core because those changes should apply to all engines. We want to catch problems as close to the source as possible. This is also a really expensive test because it spins up a catalog and Spark.

I was able to get a test to fail by adding this to the end of the `TestSortOrder.testColumnDropWithSortOrder` test case that already drops a column referenced by an old sort order:

```java

    // ensure that the table metadata can be serialized and reloaded with an invalid order
    TableMetadataParser.fromJson(TableMetadataParser.toJson(table.ops().current()));
```

That just round-trip serializes the metadata.

Reply:
Actually, here's a quick diff:

```diff
diff --git a/core/src/test/java/org/apache/iceberg/TestSortOrder.java b/core/src/test/java/org/apache/iceberg/TestSortOrder.java
index 45ff0bd71f..6b5100ed66 100644
--- a/core/src/test/java/org/apache/iceberg/TestSortOrder.java
+++ b/core/src/test/java/org/apache/iceberg/TestSortOrder.java
@@ -320,6 +320,9 @@ public class TestSortOrder {
         "Order ID must match", TableMetadata.INITIAL_SORT_ORDER_ID + 1, actualOrder.orderId());
     Assert.assertEquals(
         "Schema must have one less column", initialColSize - 1, table.schema().columns().size());
+
+    // ensure that the table metadata can be serialized and reloaded with an invalid order
+    TableMetadataParser.fromJson(TableMetadataParser.toJson(table.ops().current()));
   }
 
   @Test
```

Reply:
> We should avoid adding tests are are specific to Spark, but only test changes to core because those changes should apply to all engines.

I agree, but chasing bugs is very easy using Spark (just that's what I'm most familiar with), so this makes it also easy to distill a test from it :)

Updated the test, that's a neat trick.